src/dev/flang/tools/Fuzion.java

```diff
@@ -744,7 +781,23 @@ else if (_main != null)
       }
     if (_main == null && !_readStdin && _backend.needsMain())
       {
-        fatal("missing main feature name in command line args");
+        if (argsLeft != null && argsLeft.size() >= 1)
+          {
+            String mainOrStdin = argsLeft.remove(0);
+
+            if (mainOrStdin.equals("-"))
+              {
+                _readStdin = true;
+              }
+            else
+              {
+                _main = mainOrStdin;
+              }
+          }
+        else
```

Comment:
I am a bit unhappy to have this logic duplicated here, but I also do not see an easy solution to avoid this, so leave it for now.